src/main/java/com/amazon/opendistroforelasticsearch/sql/utils/SQLFunctions.java

```diff
@@ -710,6 +746,51 @@ private String upper(String property, String culture) {
         return new Tuple<>(name, def(name, "(int) " + getPropertyOrStringValue(field) + ".charAt(0)"));
     }
 
+    private Tuple<String, String> iif(List<KVValue> paramers) {
+        String expr1 = paramers.get(1).value.toString();
+        String expr2 = paramers.size() > 2 ? paramers.get(2).value.toString() : null;
+
+        if (paramers.get(0).value instanceof MethodField) {
+            String condition = getScriptText((MethodField) paramers.get(0).value);
+            return iif(condition, expr1, expr2);
+        } else if (paramers.get(0).value instanceof SQLBooleanExpr) {
+            Boolean condition = ((SQLBooleanExpr) paramers.get(0).value).getValue();
+            return iif(condition, expr1, expr2);
+        } else {
+            // (KVValue instance) parameters.get(0) is the parsed condition: key (String) == value (Object)
+            String key = getPropertyOrValue(paramers.get(0).key);
+            String value = getPropertyOrValue(paramers.get(0).value.toString());
+            String condition = key + " == " + value;
+            return iif(condition, expr1, expr2);
+        }
+    }
+
```

Comment:
Detailed explanation of cases where the condition is in the format of `a=b`:

`a` is parsed as the key (String) of a KVValue (get from `paramers.get(0)`), and `b` is parsed as the value (Object) of this KVValue.

Either `a` or `b` could be a column name, literal, or a number:
* if `isNumeric` is true --> number
* else if this string is single quoted --> literal
* else --> column name

Reply:
I recommed to put it as a comment into code itself :)

Reply:
+1